src/api/transactionsHistory.ts

```diff
@@ -18,40 +18,52 @@ async function extractTransactionsHistory(
         first: 1000,
         skip: 0,
     };
-    const responseActions = await connector.graphqlRequest({
-        type: 'graph',
-        op: 'AllocatorActions',
-        query: `
-    query AllocatorActions( $skip: Int! $first: Int! $where: AllocatorAction_filter) 
-        { 
-        allocatorActions( skip: $skip, first: $first, orderBy: createdAt, orderDirection: desc, where: $where, ) 
-        { id assets createdAt actionType }}
-    `,
-        variables: vars_getActions,
-        onSuccess: function (value: Response): Response {
-            return value;
-        },
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        onError: function (reason: any): PromiseLike<never> {
-            throw new Error(`Failed to get vault from Stakewise: ${reason}`);
-        },
-    });
+    try {
+        const responseActions = await connector.graphqlRequest({
+            type: 'graph',
+            op: 'AllocatorActions',
+            query: `
+        query AllocatorActions( $skip: Int! $first: Int! $where: AllocatorAction_filter) 
+            { 
+            allocatorActions( skip: $skip, first: $first, orderBy: createdAt, orderDirection: desc, where: $where, ) 
+            { id assets createdAt actionType }}
+        `,
+            variables: vars_getActions,
+            onSuccess: function (value: Response): Response {
+                return value;
+            },
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            onError: function (reason: any): PromiseLike<never> {
+                throw new Error(`Failed to get vault from Stakewise: ${reason}`);
+            },
+        });
+
+        const actionsData = await responseActions.json();
```

Comment:
The check for `.ok` is missing from here (also this and next call needs to be extracted into some common method, see comment https://github.com/ChorusOne/opus-pool-sdk/pull/16/files#r1486368244)